### ISSUE TITLE
Sessions not downloading on the HoloLens 1 or 2

### ISSUE THREAD

**Issue by wekitecs** [WEKIT-ECS/MIRAGE-XR#111]
In GitLab by @robhillman97 on Sep 28, 2020, 15:18

In the unity editor some sessions on the start menu can be clicked and then successfully downloaded, shown by a tick over the cloud symbol. When deployed onto the HoloLens 1 or 2 all sessions are not correctly downloading and instead show a red cross over the cloud symbol.

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Sep 28, 2020, 15:32

Have you tried downloading older sessions? I think this is an issue with the recorder upload. In the editor, the most recent sessions fail downloading because the player cannot read the zip file but older sessions are ok. Also see #107 for more info on this.

**Comment by wekitecs** [comment]
In GitLab by @robhillman97 on Sep 28, 2020, 15:41

Yeah I have tried with a few sessions old and new, I have also tried downloading sessions that I know can be downloaded and played in unity but they still do not download when using the HoloLens. I am re-building and deploying at the moment in developer mode so hopefully the console window will help me find out why this is happening.

**Comment by wekitecs** [comment]
In GitLab by @robhillman97 on Sep 28, 2020, 17:20

when attempting to download a session the following exceptions are thrown in the visual studio debug window: 

Exception thrown at 0x00007FF9126439EC in WEKIT.ONE.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x000000B082BFED00.

Exception thrown at 0x00007FF9126439EC in WEKIT.ONE.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x000000B082BFF180.

Exception thrown at 0x00007FF9126439EC in WEKIT.ONE.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x000000B082BFF180.

Exception thrown at 0x00007FF9126439EC in WEKIT.ONE.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x000000B082BFEAA0.

Exception thrown at 0x00007FF9126439EC in WEKIT.ONE.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x000000B082BFEAE0.

Exception thrown at 0x00007FF9126439EC in WEKIT.ONE.exe: Microsoft C++ exception: Il2CppExceptionWrapper at memory location 0x000000B082BFE370.

I will look into this and try and find out why this happens

**Comment by wekitecs** [comment]
In GitLab by @robhillman97 on Sep 28, 2020, 18:38

the debug window also displays the following:

onecore\com\combase\dcomrem\preventrundownbias.cpp(1310)\combase.dll!00007FF912C06578: (caller: 00007FF912C063B8) LogHr(6) tid(ad4) 80070005 Access is denied.

a few lines after the thrown exceptions once a session download is attempted

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Oct 1, 2020, 10:31

I added an in-application console view and it seems like a script (most likely the zip utilities) is trying to create a directory in C:/ and has no right to do this on the HoloLens. Probably one of the file paths is not set up correctly for the HoloLens.

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Oct 1, 2020, 10:48

![image](https://platform.xr4all.eu/wekit-ecs/mirage-xr/uploads/18384dde808727e05537db71e5b19d36/image.png)

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Oct 1, 2020, 12:55

I found a fix for it: The zip extractor makes sure that the target folder (which is the app data folder) exists by creating it. In the editor, this works since the PC gives more permissions to the editor but on the HoloLens, the app is only allowed to write into the app data folder. I added a check if the folder exists (which is should by default) and skip the directory creation.
